src/ItemGrid.h

```diff
@@ -186,12 +184,25 @@ class cItemGrid
 	int     m_Width;
 	int     m_Height;
 	int     m_NumSlots;  // m_Width * m_Height, for easier validity checking in the access functions
-	cItem * m_Slots;     // x + m_Width * y
+	mutable std::unique_ptr<cItem[]> m_Slots;  // x + m_Width * y, DO NOT USE DIRECTLY -- MAY BE NULLPTR
 
 	cListeners       m_Listeners;    ///< Listeners which should be notified on slot changes; the pointers are not owned by this object
 	cCriticalSection m_CSListeners;  ///< CS that guards the m_Listeners against multi-thread access
 	bool             m_IsInTriggerListeners;  ///< Set to true while TriggerListeners is running, to detect attempts to manipulate listener list while triggerring
 
+
+	/** Access to the slots. Always use this instead of m_Slots directly.
+	Will cause allocation if IsCompletelyEmpty() was true before the call. */
+	cItem * GetSlots();
+	const cItem * GetSlots() const;
+
+	/** Returns true if no storage has been allocated yet. */
+	bool IsCompletelyEmpty() const;
```

Comment:
Because it's possible for storage to be allocated but all the individual `cItem`s are empty.  I would describe that as empty but `IsCompletelyEmpty` would be false.

I'm not really happy with the name though. Maybe `IsStorageAllocated` or something would be better?

Reply:
Jep, sounds way better than the old one! The IsEmpty came from my idea, that if the container is empty, we could deallocate the space.